src/game/client/draw/w3dlaserdraw.cpp

```diff
@@ -13,3 +13,314 @@
+{
+}
+
+void W3DLaserDrawModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    static FieldParse dataFieldParse[] = {
+        { "NumBeams", &INI::Parse_Unsigned_Int, nullptr, offsetof(W3DLaserDrawModuleData, m_numBeams) },
+        { "InnerBeamWidth", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_innerBeamWidth) },
+        { "OuterBeamWidth", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_outerBeamWidth) },
+        { "InnerColor", &INI::Parse_Color_Int, nullptr, offsetof(W3DLaserDrawModuleData, m_innerColor) },
+        { "OuterColor", &INI::Parse_Color_Int, nullptr, offsetof(W3DLaserDrawModuleData, m_outerColor) },
+        { "MaxIntensityLifetime",
+            &INI::Parse_Duration_Unsigned_Int,
+            nullptr,
+            offsetof(W3DLaserDrawModuleData, m_maxIntensityFrames) },
+        { "FadeLifetime", &INI::Parse_Duration_Unsigned_Int, nullptr, offsetof(W3DLaserDrawModuleData, m_fadeFrames) },
+        { "Texture", &INI::Parse_AsciiString, nullptr, offsetof(W3DLaserDrawModuleData, m_textureName) },
+        { "ScrollRate", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_scrollRate) },
+        { "Tile", &INI::Parse_Bool, nullptr, offsetof(W3DLaserDrawModuleData, m_tile) },
+        { "Segments", &INI::Parse_Unsigned_Int, nullptr, offsetof(W3DLaserDrawModuleData, m_segments) },
+        { "ArcHeight", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_arcHeight) },
+        { "SegmentOverlapRatio", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_segmentOverlapRatio) },
+        { "TilingScalar", &INI::Parse_Real, nullptr, offsetof(W3DLaserDrawModuleData, m_tilingScalar) },
+    };
+
+    ModuleData::Build_Field_Parse(p);
+    p.Add(dataFieldParse, 0);
+}
+
+W3DLaserDraw::W3DLaserDraw(Thing *thing, ModuleData const *module_data) :
+    DrawModule(thing, module_data), m_line3D(nullptr), m_texture(nullptr), m_lineTransparency(1.0f), m_setLaserPosition(true)
+{
+    const W3DLaserDrawModuleData *data = Get_Module_Data();
+    m_texture = W3DAssetManager::Get_Instance()->Get_Texture(data->m_textureName);
+
+    if (m_texture) {
+        if (!m_texture->Is_Initialized()) {
+            m_texture->Init();
+        }
+
+        SurfaceClass::SurfaceDescription desc;
+        m_texture->Get_Level_Description(desc, 0);
+        m_lineTransparency = desc.width / desc.height;
+    }
+
+    float innerred;
+    float innergreen;
+    float innerblue;
+    float inneralpha;
+    float outerred;
+    float outergreen;
+    float outerblue;
+    float outeralpha;
+    Get_Color_Components_Real(data->m_innerColor, &innerred, &innergreen, &innerblue, &inneralpha);
+    Get_Color_Components_Real(data->m_outerColor, &outerred, &outergreen, &outerblue, &outeralpha);
+    m_line3D = new SegmentedLineClass *[data->m_segments * data->m_numBeams];
+
+    for (unsigned int i = 0; i < data->m_segments; i++) {
+        for (unsigned int j = 0; j < data->m_numBeams; j++) {

```

Comment:
the loop starts from numbeams - 1

Reply:
Looking at other binaries the loop goes down in all of them so I have changed it to go down.